pkg/view/router.go

```diff
@@ -54,17 +109,220 @@ func New(c config.DashboardConfig, auth func(h http.Handler) http.Handler) (*chi
 	})
 
 	r.Get("/connections", func(w http.ResponseWriter, r *http.Request) {
-		err := gv.Render(w, http.StatusOK, "pages/connections/index", loadModel(r))
+		user, ok := getUser(r)
+		if !ok {
+			http.Error(w, "User not found", http.StatusInternalServerError)
+			return
+		}
+		destinations, err := storageServices.Database.GetDestinations(r.Context(), user.ID)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		m := loadModel(r)
+		m.Connections = Connections{
+			Destinations: destinations,
+		}
+		err = gv.Render(w, http.StatusOK, "pages/connections/index", m)
 		if err != nil {
 			http.Error(w, err.Error(), http.StatusInternalServerError)
 		}
 	})
 
 	r.Get("/connections/new", func(w http.ResponseWriter, r *http.Request) {
-		err := gv.Render(w, http.StatusOK, "pages/connections/new", loadModel(r))
+		err := gv.Render(w, http.StatusOK, "pages/connections/upsert", loadModel(r))
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+	})
+
+	r.Post("/connections/upsert", func(w http.ResponseWriter, r *http.Request) {
+		user, ok := getUser(r)
+		if !ok {
+			http.Error(w, "User not found", http.StatusInternalServerError)
+			return
+		}
+
+		err := r.ParseForm()
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		id := r.Form.Get("id")
+		if id != "" {
+			http.Error(w, "Editing connections not yet supported", http.StatusBadRequest)
+			return
+		}
+
+		settings := map[string]any{}
+		name := r.Form.Get("name")
+
+		t := r.Form.Get("type")
+		switch t {
+		case "duckdb":
+			// XXX breadchris support file destination?
+
+			tok := r.Form.Get("token")
+			if tok == "" {
+				http.Error(w, "Must specify a token", http.StatusBadRequest)
+				return
+			}
+			db := r.Form.Get("database")
+			if db == "" {
+				http.Error(w, "Must specify a token", http.StatusBadRequest)
+				return
+			}
+
+			// XXX breadchris what validation should be done here?
+
+			settings["token"] = tok
+			settings["database"] = db
+		default:
+			http.Error(w, "Unknown connection type", http.StatusBadRequest)
+			return
+		}
+
+		cd := config.Destination{
+			Type:     t,
+			Settings: settings,
+		}
+
+		err = destManager.TestCredentials(cd)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
```

Comment:
It would be way better to flash an error message, and then pre-populate the form with whatever fields they entered. I think gorilla will do message flashing: https://github.com/gorilla/sessions

Here's the tailwind code for an alert:

```
<div class="rounded-md bg-red-50 p-4">
  <div class="flex">
    <div class="flex-shrink-0">
      <svg class="h-5 w-5 text-red-400" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true">
        <path fill-rule="evenodd" d="M10 18a8 8 0 100-16 8 8 0 000 16zM8.28 7.22a.75.75 0 00-1.06 1.06L8.94 10l-1.72 1.72a.75.75 0 101.06 1.06L10 11.06l1.72 1.72a.75.75 0 101.06-1.06L11.06 10l1.72-1.72a.75.75 0 00-1.06-1.06L10 8.94 8.28 7.22z" clip-rule="evenodd" />
      </svg>
    </div>
    <div class="ml-3">
      <h3 class="text-sm font-medium text-red-800">There were 2 errors with your submission</h3>
      <div class="mt-2 text-sm text-red-700">
        <ul role="list" class="list-disc space-y-1 pl-5">
          <li>Your password must be at least 8 characters</li>
          <li>Your password must include at least one pro wrestling finishing move</li>
        </ul>
      </div>
    </div>
  </div>
</div>
```